pkgs/by-name/oo/ooklaserver/package.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchurl
+, unzip
+, autoPatchelfHook
+}:
+let
+  filenameMapping = {
+    "x86_64-linux" = "OoklaServer-linux-x86_64-static-musl.zip";
+    # Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
+    "x86_64-darwin" = "OoklaServer-macosx.zip";
+    "aarch64-darwin" = "OoklaServer-macosx.zip";
+    "aarch64-linux" = "OoklaServer-linux-aarch64-static-musl.zip";
+    "i686-windows" = "OoklaServer-windows32.zip";
+    "x86_64-windows" = "OoklaServer-windows64.zip";
+    # OoklaServer-linux64-deb9.zip
+    # OoklaServer-freebsd12_64.zip
+    # OoklaServer-freebsd13_64.zip
+  };
+in
+stdenv.mkDerivation rec {
+  version = "2.11.1.2";
+  pname = "OoklaServer";
+
+  src = fetchurl {
+    url = "https://web.archive.org/web/20240703022648/https://install.speedtest.net/ooklaserver/stable/OoklaServer.tgz";
+    hash = "sha256-tctLtTGmrVHs+4pI1PRHrqY+a4ISs6TKvLRKlFdWw88=";
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ unzip autoPatchelfHook ];
+
+  unpackPhase = ''
+    tar -xvf $src
+    unzip ${filenameMapping.${stdenv.system}}
```

Comment:
Shall we move this to `installPhase` instead?

Reply:
It is more like unpack the right version, but I am open to that